samples/net/sockets/http_get/src/http_get.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+
+#ifndef __ZEPHYR__
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <netdb.h>
+
+#else
+
+#include <net/socket.h>
+#include <kernel.h>
+#include <net/net_app.h>
+
+#endif
+
+#define SSTRLEN(s) (sizeof(s) - 1)
+#define CHECK(r) if (r == -1) { printf("Error: " #r "\n"); }
+
+#define REQUEST "GET / HTTP/1.0\r\n\r\n"
+
+static char response[1024];
+
+void dump_addrinfo(const struct addrinfo *ai) {
+	printf("addrinfo @%p: fam=%d, socktype=%d, proto=%d, "
+	       "addr_fam=%d, addr_port=%x\n",
+	       ai, ai->ai_family, ai->ai_socktype, ai->ai_protocol,
+	       ai->ai_addr->sa_family,
+	       ((struct sockaddr_in*)ai->ai_addr)->sin_port);
+}
+
+int main(void)
+{
+	static struct addrinfo hints;
+	struct addrinfo *res;
+	int st, sock;
+
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	st = getaddrinfo("google.com", "80", &hints, &res);
```

Comment:
This seems to require direct internet access which is ok, but prevents this application to be used in closed test environments. We might need to consider changing this at some point.

Reply:
Right, this is more a "sample" than a "test", so made to show some "real-world" effect. I added a README to the patch, which explicitly states the Internet connection requirement, and also mentions that the code can be manually edited to connect to another server (e.g. on a local net).